### PR TITLE
 automate-backend-elasticsidecar changes 

### PR DESCRIPTION
**Pull request by FaizanSRE**
Signed-off-by: FaizanSRE <ffulara@msystechnologies.com>

### :nut_and_bolt: Description: What code changed, and why?
For automate-backend-elasticsidecar, we have done the changes.
New Packages Changes
Changes is all about managing the dependency via gemlock file. please refer the PR link
Old Package
we are managing the deps in plan.sh.
In new approach we are doing the bundle install, which place a gems at different location,
because of that we are getting the error.
please refer the below screenshot

image
For a time being we are reverting the changes. we are using previous approach to unblock the testing for now.
And we are creating the ticket in backlog to manage the deps via gem file.

### :white_check_mark: Checklist

**All PRs** must tick these:

- [x] I have read the [CONTRIBUTING document](https://github.com/chef/automate/blob/master/CONTRIBUTING.md).
- [x] All commits signed-off for the [Developer Certification of Origin](https://github.com/chef/chef/blob/master/CONTRIBUTING.md#developer-certification-of-origin-dco).

With occasional exceptions, all PRs **from Progress employees** must tick these:

- [x] Is the code clear? *(complicated code or lots of comments--subdivide and use well-named methods, meaningful variable names, etc.)*
- [ ] Consistency checked? *(user notifications, user prompts, visual patterns, code patterns, variable names)*
- [ ] Repeated code blocks eliminated? *(adapt and reuse existing components, blocks, functions, etc.)*
- [ ] Spelling, grammar, typos checked? *(at a minimum use `make spell` in any component directory)*
- [ ] Code well-formatted? *(indents, line breaks, etc. improve rather than hinder readability)*

All PRs **from Progress employees** should tick these if appropriate:

- [ ] Tests added/updated? (all new code needs new tests)
- [ ] Docs added/updated? (all customer-facing changes)

*Please add a note next to any checkbox above if you are NOT ticking it.*

### CODE REVIEW

**Review by punitmundra — APPROVED** [review]

File: components/automate-backend-elasticsidecar/habitat/bin/elastic_sidecar.rb [hunk 1]
```diff
@@ -8,7 +8,6 @@
 require 'toml-rb'
 require 'yaml'
 require 'mixlib/shellout'
-require "chefstyle"
```
Comment (kalroy) [inline components/automate-backend-elasticsidecar/habitat/bin/elastic_sidecar.rb:11]: Why are we removing this. It looks like the style validator as per https://github.com/chef/chefstyle

File: components/automate-backend-elasticsidecar/habitat/hooks/run [hunk 2]
```diff
@@ -2,9 +2,6 @@
 
 exec 2>&1
 
-# Required to allow common name feild in certificate. Feature soon to deprecated by 1.17
```
Comment (kalroy) [inline components/automate-backend-elasticsidecar/habitat/hooks/run:5]: Why are we removing this? I can see it being used in all the plan.sh

**Review by kalroy — CHANGES_REQUESTED** [review]
There is no issue as such. But have some questions which needs clarifications

File: components/automate-backend-elasticsidecar/habitat/bin/elastic_sidecar.rb (hunk 1 above)
Comment (FaizanSRE) [reply components/automate-backend-elasticsidecar/habitat/bin/elastic_sidecar.rb:11]: As we were not doing any style lintment as per chef-style, that is why we have removed that gem

Comment (kalroy) [reply components/automate-backend-elasticsidecar/habitat/bin/elastic_sidecar.rb:11]: Is it something agreed upon with other Ruby developers?

**Review by kalroy — CHANGES_REQUESTED** [review]

File: components/automate-backend-elasticsidecar/habitat/bin/elastic_sidecar.rb (hunk 1 above)
Comment (FaizanSRE) [reply components/automate-backend-elasticsidecar/habitat/bin/elastic_sidecar.rb:11]: @rishichawda Can you have a look at this ?

Comment (tas50) [reply components/automate-backend-elasticsidecar/habitat/bin/elastic_sidecar.rb:11]: Long term we should be linting everything Ruby we produce as a org. All projects in the org *except* Automate are doing that.

Comment (FaizanSRE) [reply components/automate-backend-elasticsidecar/habitat/bin/elastic_sidecar.rb:11]: Sure will get it done, we have created ticket for the same https://app.zenhub.com/workspaces/the-delta-quadrant-5fc7c4c3922dc10021323fa6/issues/chef/automate/6076

Comment (rishichawda) [reply components/automate-backend-elasticsidecar/habitat/bin/elastic_sidecar.rb:11]: Yep. chefstyle is used for all chef oss projects so we should probably keep it.

Comment (FaizanSRE) [reply components/automate-backend-elasticsidecar/habitat/bin/elastic_sidecar.rb:11]: I have added chefstyle

**Review by kalroy — APPROVED** [review]

**Review by rishichawda — APPROVED** [review]

**Pull request merged**